src/jaxoplanet/experimental/starry/maps.py

```diff
@@ -0,0 +1,128 @@
+    period: Optional[float] = None
+
+    def __init__(
+        self,
+        *,
+        y: Optional[Ylm] = None,
+        inc: Optional[float] = None,
+        obl: Optional[float] = None,
+        u: Optional[tuple] = None,
+        period: Optional[float] = None,
+    ):
+
+        if y is None:
+            y = Ylm({(0, 0): 1})
+        if inc is None:
+            inc = jnp.pi / 2
+        if obl is None:
+            obl = 0.0
+        if u is None:
+            u = ()
+
+        self.y = y
+        self.inc = inc
+        self.obl = obl
+        self.u = u
+        self.period = period
+
+    @property
+    def poly_basis(self):
+        return jax.jit(poly_basis(self.deg))
+
+    @property
+    def udeg(self):
+        return len(self.u)
+
+    @property
+    def ydeg(self):
+        return self.y.ell_max
+
+    @property
+    def deg(self):
+        return self.ydeg + self.udeg
+
+    def render(self, theta=0, res=400):
+        _, xyz = ortho_grid(res)
+        pT = self.poly_basis(*xyz)
+        Ry = left_project(self.ydeg, self.inc, self.obl, theta, 0.0, self.y.todense())
+        A1Ry = A1(self.ydeg) @ Ry
+        p_y = Pijk.from_dense(A1Ry, degree=self.ydeg)
+        U = np.array([1, *self.u])
+        p_u = Pijk.from_dense(U @ U0(self.udeg), degree=self.udeg)
+        p = (p_y * p_u).todense()
+        return jnp.reshape(pT @ p, (res, res))
+
+    def show(self, theta=0, res=400, graticule=6, **kwargs):
+        import matplotlib.pyplot as plt
+
+        plt.imshow(
+            self.render(theta, res), origin="lower", **kwargs, extent=(-1, 1, -1, 1)
+        )
+        if graticule is not None:
+            self.graticule(theta)
+        plt.axis(False)
+
+    def graticule(self, theta=0, pts=100, **kwargs):
+        import matplotlib.pyplot as plt
+
+        kwargs.setdefault("c", kwargs.pop("color", "k"))
+        kwargs.setdefault("lw", kwargs.pop("linewidth", 1))
+        kwargs.setdefault("alpha", 0.5)
+
+        # plot lines
+        lat, lon = lon_lat_lines(pts=pts)
+        lat = rotate_lines(lat, self.inc, self.obl, theta)
+        plot_lines(lat, **kwargs)
+        lon = rotate_lines(lon, self.inc, self.obl, theta)
+        plot_lines(lon, **kwargs)
+        theta = np.linspace(0, 2 * np.pi, 2 * pts)
+        plt.plot(np.cos(theta), np.sin(theta), **kwargs)
+
+    def light_curve(self, system, time):
+        theta = 0.0 if self.period is None else 2.0 * np.pi * time / self.period
+        lc_func = jnp.vectorize(
+            partial(
+                light_curve,
+                self.ydeg,
+                self.y.todense(),
+                self.u,
+                self.inc,
+                self.obl,
+                theta,
+            ),
+            signature="(),(),(),()->()",
+        )
+
+        def body_lc(body):
+            x, y, z = body.relative_position(time)
+            return lc_func(x.magnitude, y.magnitude, z.magnitude, body.radius.magnitude)
+
+        lc = jax.vmap(body_lc)(system._body_stack)
```

Comment:
This won't work in general because only a Keplerian `System` has a `_body_stack`. In [`LimbDarkLightCurve`](https://github.com/exoplanet-dev/jaxoplanet/blob/02f55c63536ccce24a3fe8d3c392133f79dc41d1/src/jaxoplanet/light_curves.py#L25), we extract the relevant quantities manually and restrict the allowed input systems using the [`LightCurveOrbit`](https://github.com/exoplanet-dev/jaxoplanet/blob/02f55c63536ccce24a3fe8d3c392133f79dc41d1/src/jaxoplanet/proto.py#L14) protocol, but I don't think that's a good solution.

Probably what we want to do is have a common API for mapping functions over bodies in orbits. This is related to some of the things @soichiro-hattori has been thinking about wrt exposure time integration. Maybe it's ok to leave this as it is, and think about what a better abstraction would be later.

Reply:
I kept this one for a later PR. for now I just deleted the `light_curve` function